types/styled-react-modal/index.d.ts

```diff
@@ -1,37 +1,79 @@
+         */
+        isOpen: boolean;
+        /**
+         * When true, scrolling in the document body is not disabled when the modal is open.
+         */
+        allowScroll?: boolean | undefined;
+        /**
+         * A props object that is spread over the `backgroundComponent` when included.
+         */
+        backgroundProps?: object | undefined;
+        /**
+         * A function that is called after the modal opens.
+         */
+        afterOpen?: (() => void) | undefined;
+        /**
+         * A function that is called after the modal closes.
+         */
+        afterClose?: (() => void) | undefined;
+        /**
+         * A function that is called before the modal opens.
+         * If this function returns a promise, then the modal is opened after the promise is resolved.
+         */
+        beforeOpen?: (() => void | Promise<void>) | undefined;
+        /**
+         * A function that is called before the modal closes.
+         * If this function returns a promise, then the modal is closed after the promise is resolved.
+         */
+        beforeClose?: (() => void | Promise<void>) | undefined;
+        /**
+         * A function that is called when the escape key is pressed while the modal is open.
+         */
+        onEscapeKeydown?: ((event: React.KeyboardEvent) => void) | undefined;
+        /**
+         * A function that is called when the modal background is clicked.
+         */
+        onBackgroundClick?: ((event: React.MouseEvent) => void) | undefined;
+    }
 
-declare class Modal extends React.Component<ModalProps> {
-    static styled(
-        object: CSSObject | InterpolationFunction<any>,
-    ): StyledComponent<React.ComponentClass<ModalProps>, any>;
-    static styled(
-        strings: TemplateStringsArray,
-        ...interpolations: any[]
-    ): StyledComponent<React.ComponentClass<ModalProps>, any>;
-}
+    interface ModalProviderProps {
+        children: React.ReactNode;
+        /**
+         * A styled component to be used as the default modal background.
+         * If not provided, library defaults will be used.
+         *
+         * @default BaseModalBackground
+         */
+        backgroundComponent?: AnyStyledComponent | undefined;
+    }
 
-interface ModalProviderProps {
-    backgroundComponent?: AnyStyledComponent | undefined;
-    children: React.ReactNode;
+    class ModalProvider extends React.Component<ModalProviderProps> {}
 }
 
-declare class ModalProvider extends React.Component<ModalProviderProps> {}
+declare class StyledReactModel extends React.Component<StyledReactModel.ModalProps> {
+    /**
+     * Factory method that accepts a tagged template literal and returns a `<Modal>` component
+     * with styles included.
+     */
+    static styled: (
+        ...args: Parameters<ThemedStyledFunctionBase<"div", any>>
+    ) => typeof StyledReactModel;
+}
 
-export default Modal;
-export { BaseModalBackground, ModalProvider };
+export = StyledReactModel;
```

Comment:
What happened to the `default` export? Why is this changed to `export =` ? ESM modules should generally not use `export =`.

Reply:
Done. I thought it is a cjs module back when I opened the PR. ☹️